### PR TITLE
Move childMap, latestMessages, blockLookup usages into BlockDag 

### PR DESCRIPTION
**Pull request by KentShikama**
## Overview
Provide a brief description of what this PR does, and why it's needed.

Relative to #656 . This is part of a larger project to make the entire codebase functional.

### Does this PR relate to an RChain JIRA issue? 
If applicable, add link to corresponding JIRA issue.

### Complete this checklist before you submit the PR
- [X] This PR contains no more than 200 lines of code, excluding test code.
- [X] This PR meets [RChain development coding standards](https://rchain.atlassian.net/wiki/spaces/DOC/pages/28082177/Coding+Standards).
- [X] You have someone in mind to assign for review. Make this assignment after submitting the PR.

### Notes
Optional. Add any notes on caveats, approaches you tried that didn't work, or anything else.

### CODE REVIEW

File: casper/src/main/scala/coop/rchain/casper/Casper.scala [hunk 1]
```diff
@@ -53,19 +54,13 @@ sealed abstract class MultiParentCasperInstances {
...
       type Validator = ByteString
 
-      private val _childMap: mutable.HashMap[BlockHash, mutable.HashSet[BlockHash]] =
-        new mutable.HashMap[BlockHash, mutable.HashSet[BlockHash]]()
-      private val _latestMessages: mutable.HashMap[Validator, BlockHash] =
-        new mutable.HashMap[Validator, BlockHash]()
-      private val blockLookup: mutable.HashMap[BlockHash, BlockMessage] =
-        new mutable.HashMap[BlockHash, BlockMessage]()
+      var blockDag: BlockDag = BlockDag().copy(
+        blockLookup = HashMap[BlockHash, BlockMessage](genesis.blockHash -> genesis))
```
Comment (KentShikama) [inline casper/src/main/scala/coop/rchain/casper/Casper.scala]: This will eventually be pushed up into MonadState

File: casper/src/main/scala/coop/rchain/casper/BlockDag.scala [hunk 2]
```diff
@@ -0,0 +1,31 @@
+package coop.rchain.casper
+
+import com.google.protobuf.ByteString
```
Comment (birchmd) [inline casper/src/main/scala/coop/rchain/casper/BlockDag.scala]: Remove this import since you are using the `BlockHash` and `Validator` aliases.

File: casper/src/main/scala/coop/rchain/casper/BlockDag.scala [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+import coop.rchain.casper.Estimator.{BlockHash, Validator}
+import coop.rchain.casper.protocol.BlockMessage
+
+import scala.collection.immutable.{HashMap, HashSet}
+
+final case class BlockDag(idToBlocks: HashMap[Int, BlockMessage],
+                          blockLookup: HashMap[ByteString, BlockMessage],
```
Comment (birchmd) [inline casper/src/main/scala/coop/rchain/casper/BlockDag.scala]: Should be `blockLookup: HashMap[BlockHash, BlockMessage]`

File: casper/src/main/scala/coop/rchain/casper/BlockDag.scala [hunk 4]
```diff
@@ -0,0 +1,31 @@
...
+                          childMap: HashMap[BlockHash, HashSet[BlockHash]],
+                          latestMessages: HashMap[Validator, BlockHash],
+                          currentId: Int)
+
+object BlockDag {
+  def apply(idToBlocks: HashMap[Int, BlockMessage],
```
Comment (birchmd) [inline casper/src/main/scala/coop/rchain/casper/BlockDag.scala]: This isn't necessary to provide explicitly because it would be auto-generated by the compiler since `BlockDag` is a case class.

**Comment by KentShikama** [comment]
@birchmd Ready for another review

File: casper/src/main/scala/coop/rchain/casper/Estimator.scala [hunk 5]
```diff
@@ -17,24 +17,25 @@ object Estimator {
...
         blocks
       } else {
-        sortChildren(newBlocks, scores)
+        sortChildren(newBlocks, childMap, scores)
       }
     }
-    def replaceBlockHashWithChildren(b: BlockHash) = {
+    def replaceBlockHashWithChildren(childMap: HashMap[BlockHash, HashSet[BlockHash]],
+                                     b: BlockHash) = {
       val empty                 = new HashSet[BlockHash]()
```
Comment (birchmd) [inline casper/src/main/scala/coop/rchain/casper/Estimator.scala]: I know not part of this PR, but it's something that I must have missed before. This should be `HashSet.empty[BlockHash]`. There is only one empty immutable HashSet, (and HashMap too), so we should not instantiate it with `new`.

**Comment by KentShikama** [comment]
@birchmd PTAL
